tests/test_slim_engine.py

```diff
@@ -421,6 +423,70 @@ def count_mut_types(ts):
             ).split()
             with pytest.raises(SystemExit):
                 capture_output(stdpopsim.cli.stdpopsim_main, cmd)
+        # test DFE with demography and interval as a bed file
+        lines = ["chr1  100000  145000", "chr1  150000  302425"]
+        bedfile = open("ex.bed", "w")
+        for lin in lines:
+            bedfile.write(lin + "\n")
+        bedfile.close()
+        with tempfile.NamedTemporaryFile(mode="w") as f:
+            cmd = (
+                f"-q -e slim --slim-scaling-factor 20 --slim-path {slim_path} "
+                f"HomSap -c chr22 -s 1234 -l0.01 -o {f.name} "
+                "-d OutOfAfrica_3G09 --dfe Gamma_K17 "
+                "--dfe-bed-file ex.bed 10"
+            ).split()
+            capture_output(stdpopsim.cli.stdpopsim_main, cmd)
+            ts = tskit.load(f.name)
+        assert ts.num_samples == 10
+        assert all(tree.num_roots == 1 for tree in ts.trees())
+        n_mut_types = count_mut_types(ts)
+        assert n_mut_types[0] > 0
+        assert n_mut_types[1] > 0
```

Comment:
We should check using top-level metadata reported from SLiM whether the intervals are doing what we think they should here. Want me ot take a crack at this?

Reply:
sure, go ahead :)

Reply:
ok, maybe let's get this PR merged and open an issue about that.